ibis/pandas/execution/window.py

```diff
@@ -169,6 +179,9 @@ def trim_with_timecontext(data, timecontext: Optional[TimeContext]):
     # indexes are retturned.
     subset = df.loc[df[TIME_COL].between(*timecontext)]
 
+    # re-indexing index to count from 0
+    subset = subset.reset_index(drop=True).reset_index()
```

Comment:
Because we have a filter option `between` applied on subset. So the first `reset_index` may produce a Series of index (3,4,5....) We will need to re-index it to count from 0